aderyn_core/src/detect/low/state_variables_not_logged_in_event.rs

```diff
@@ -0,0 +1,237 @@
+                    let mut counter: u32 = 0; //counter for number of times state variable is logged in events in same function
+
+                    let body = function
+                        .body
+                        .clone()
+                        .ok_or("Failed to clone function body: body is None")?;
+                    //now we must search all events listed in function
+                    for event_emission in ExtractEmitStatements::from(&body).extracted.into_iter() {
+                        let children = event_emission
+                            .event_call
+                            .children(context)
+                            .ok_or("Failed to unwrap ASTNode children: children is None")?;
+
+                        //if a parameter name matches the LHS or RHS name, increment a counter
+                        for child in children.into_iter() {
+                            if check_event_args(
+                                child,
+                                &left_identifier_name,
+                                &right_identifier_name,
+                                referenced_id,
+                            ) {
+                                counter += 1;
+                            }
+                        }
+                        //if the counter is still zero then no event logs the state var and we capture the assignment line
+                        //reset the counter to zero for each loop of a new state var
+                    }
+                    if counter == 0 {
+                        //no event mentions the state variable, report as an issue
+                        capture!(self, context, assignment);
+                    }
+                }
+            }
+        }
+
+        Ok(!self.found_instances.is_empty())
+    }
+
+    fn title(&self) -> String {
+        String::from("State variable modification not logged in event")
+    }
+
+    fn description(&self) -> String {
+        String::from("State variable assignment not recorded in event logs, this will make it difficult for off-chain applications to track changes to state or searching historic event logs")
+    }
+
+    fn severity(&self) -> IssueSeverity {
+        IssueSeverity::Low
+    }
+
+    fn instances(&self) -> BTreeMap<(String, usize, String), NodeID> {
+        self.found_instances.clone()
+    }
+
+    fn name(&self) -> String {
+        format!(
+            "{}",
+            IssueDetectorNamePool::StateVariableNotLoggedInEventDetector
+        )
+    }
+}
+
+//expressions can be identifiers for variables and memberAccess for struct subfields so we match accordingly
+fn return_identifier_referenced_dec(expression: Expression) -> Result<i64, &'static str> {
+    match expression {
+        Expression::Identifier(left_identifier) => {
+            if let Some(reference_id) = left_identifier.referenced_declaration {
+                return Ok(reference_id);
```

Comment:
Some linting issues here. Try running `cargo clippy` and resolve these, then `cargo fmt` to format properly.

Reply:
Fixed in 8caf6e8